index.test.js

```diff
@@ -59,6 +60,8 @@ describe('Deferred', () => {
       const deferred = new Deferred();
 
       assert.equal(deferred.reject('rejection-value'), deferred.promise);
+
+      return deferred.promise.catch(() => {});
```

Comment:
The node test runner is pretty sophisticated, and can more effectively sniff out tests with code which does stuff after the the test finishes. The fix here is to make sure we return a promise.